packages/scanner/src/cli/scan/watchScan.ts

```diff
@@ -7,46 +7,69 @@ import { formatReport } from './formatReport';
 import { default as buildScanner } from './scanner';
 import { exists } from 'fs';
 import { promisify } from 'util';
+import { parseConfigFile } from '../../configuration/configurationProvider';
+import assert from 'assert';
 
 type WatchScanOptions = {
   appmapDir: string;
-  configData: Configuration;
+  configFile: string;
 };
 
 export class Watcher {
-  watcher?: chokidar.FSWatcher;
+  config?: Configuration;
+  appmapWatcher?: chokidar.FSWatcher;
+  configWatcher?: chokidar.FSWatcher;
 
   constructor(private options: WatchScanOptions) {}
 
   async watch(): Promise<void> {
-    this.watcher = chokidar.watch(`${this.options.appmapDir}/**/mtime`, {
+    await this.reloadConfig();
+
+    this.configWatcher = chokidar.watch(this.options.configFile, {
+      ignoreInitial: true,
+    });
+    this.configWatcher
+      .on('add', this.reloadConfig.bind(this))
+      .on('change', this.reloadConfig.bind(this));
+
+    this.appmapWatcher = chokidar.watch(`${this.options.appmapDir}/**/mtime`, {
       ignoreInitial: true,
     });
-    this.watcher.on('add', this.scan.bind(this)).on('change', this.scan.bind(this));
+    this.appmapWatcher.on('add', this.scan.bind(this)).on('change', this.scan.bind(this));
   }
 
   abort(): void {
-    if (!this.watcher) return;
+    if (!this.appmapWatcher) return;
 
-    this.watcher.close();
-    this.watcher = undefined;
+    assert(this.configWatcher);
```

Comment:
Why are you using an assertion here and in `scan`, rather than assuring that these properties are set?

It seems especially strange that a method called `abort` is asserting anything....

Reply:
Isn’t an assertion a way to ensure that the properties are set?

Reply:
close would be a better name

Reply:
> Isn’t an assertion a way to ensure that the properties are set?
Yeah. But, under what circumstances would these properties not be set? Would you be better off ensuring that they never occurred, or is that simply not possible? If you're going to keep the `assert`, it would be clearer if you also passed it a message: https://nodejs.org/api/assert.html#assertvalue-message

> Both watchers should either be defined or undefined.
One's getting tested with an `if`, but the other is tested with an `assert`. Why are they different?

> close would be a better name
But, even if it's named `close`, it seems like it should be more defensive, and just ignore the undefined property, rather than triggering an assertion.